record/blinded_data.go

```diff
@@ -68,12 +73,33 @@ func NewNonFinalBlindedRouteData(chanID lnwire.ShortChannelID,
 	return info
 }
 
+// NewFinalHopBlindedRouteData creates the data that's provided for the final
+// hop in a blinded route.
+func NewFinalHopBlindedRouteData(constraints *PaymentConstraints,
+	pathID []byte) *BlindedRouteData {
```

Comment:
Non-blocking here, but we could also consider making them to individual types further encode the distinction in the types themselves.

Reply:
yeah good idea 👍  that would make things much nicer and easier to reason about